lib/chef/node_map.rb

```diff
@@ -269,11 +270,23 @@ def matches_version_list?(node, filters, attribute)
         end
     end
 
+    # Succeeds if:
+    # - we are in target mode, and the target_mode filter is true
+    # - we are not in target mode, and the target_mode filter is not false
+    def matches_target_mode?(filters)
+      if Chef::Config.target_mode?
+        filters[:target_mode] == true
+      else
+        !(filters[:target_mode] == false)
+      end
```

Comment:
`!!filters[:target_mode] == !!Chef::Config.target_mode?` is probably a bit clearer that you're checking boolean equality while not caring about strict true/false values and just truthy/falsey